src/constructs/acm/certificate.ts

```diff
@@ -57,43 +55,17 @@ export type GuCertificatePropsWithApp = GuDomainNameProps & AppIdentity & GuMigr
  */
 export class GuCertificate extends GuStatefulMigratableConstruct(GuAppAwareConstruct(Certificate)) {
   constructor(scope: GuStack, props: GuCertificatePropsWithApp) {
-    const hasHostedZoneId: boolean = StageAwareValue.isStageValue(props)
-      ? !!props.CODE.hostedZoneId && !!props.PROD.hostedZoneId
-      : !!props.INFRA.hostedZoneId;
+    const { app, domainName, existingLogicalId, hostedZoneId } = props;
 
-    const maybeHostedZone = !hasHostedZoneId
-      ? undefined
-      : HostedZone.fromHostedZoneId(
-          scope,
-          /* eslint-disable @typescript-eslint/no-non-null-assertion -- `hasHostedZoneId` is true, so we know `hostedZoneId` is present here */
-          AppIdentity.suffixText({ app: props.app }, "HostedZone"),
-          StageAwareValue.isStageValue(props)
-            ? scope.withStageDependentValue({
-                app: props.app,
-                variableName: "hostedZoneId",
-                stageValues: {
-                  [Stage.CODE]: props.CODE.hostedZoneId!,
-                  [Stage.PROD]: props.PROD.hostedZoneId!,
-                },
-              })
-            : props.INFRA.hostedZoneId!
-          /* eslint-enable @typescript-eslint/no-non-null-assertion */
-        );
+    const maybeHostedZone = hostedZoneId
+      ? HostedZone.fromHostedZoneId(scope, AppIdentity.suffixText({ app }, "HostedZone"), hostedZoneId)
+      : undefined;
```

Comment:
```ts
const maybeHostedZone = hostedZoneId && HostedZone.fromHostedZoneId(scope, AppIdentity.suffixText({ app }, "HostedZone"), hostedZoneId);
```

Would be preferred, however it's type is `"" | undefined | HostedZone` - I can't quite work out why the empty string is there 🤔 .

Reply:
Ah it's because "" is falsy, and `hostedZoneId` itself is `string | undefined`. I can't think of a better way than ternary here either.